src/Message/PaymentIntents/Response.php

```diff
@@ -0,0 +1,158 @@
+ */
+class Response extends BaseResponse
+{
+    /**
+     * Get the status of a payment intents response.
+     *
+     * @return string|null
+     */
+    public function getStatus()
+    {
+        if (isset($this->data['object']) && 'payment_intent' === $this->data['object']) {
+            return $this->data['status'];
+        }
+
+        return null;
+    }
+
+    /**
+     * Return true if the payment intent requires confirmation.
+     *
+     * @return bool
+     */
+    public function requiresConfirmation()
+    {
+        return $this->getStatus() === 'requires_confirmation';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getCardReference()
+    {
+        if (isset($this->data['object']) && 'payment_method' === $this->data['object']) {
+            if (!empty($this->data['id'])) {
+                return $this->data['id'];
+            }
+        }
+
+        if (isset($this->data['object']) && 'payment_intent' === $this->data['object']) {
+            if (!empty($this->data['payment_method'])) {
+                return $this->data['payment_method'];
+            }
+        }
+
+        return parent::getCardReference();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getCustomerReference()
+    {
+        if (isset($this->data['object']) && 'payment_method' === $this->data['object']) {
+            if (!empty($this->data['customer'])) {
+                return $this->data['customer'];
+            }
+        }
+
+        if (isset($this->data['object']) && 'payment_intent' === $this->data['object']) {
+            if (!empty($this->data['customer'])) {
+                return $this->data['customer'];
+            }
+        }
+
+        return parent::getCustomerReference();
+    }
+
+    /**
+     * Get the capture method of a payment intents response.
+     *
+     * @return string|null
+     */
+    public function getCaptureMethod()
+    {
+        if (isset($this->data['object']) && 'payment_intent' === $this->data['object']) {
+            return $this->data['capture_method'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getTransactionReference()
+    {
+        if (isset($this->data['object']) && 'payment_intent' === $this->data['object']) {
+            if (!empty($this->data['charges']['data'][0]['id'])) {
+                return $this->data['charges']['data'][0]['id'];
+            }
+        }
+
+        return parent::getTransactionReference();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isSuccessful()
+    {
+        if (isset($this->data['object']) && 'payment_intent' === $this->data['object']) {
+            return in_array($this->getStatus(), ['succeeded', 'requires_capture']);
+        }
+
+        return parent::isSuccessful();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isRedirect()
+    {
+        if ($this->getStatus() === 'requires_action') {
```

Comment:
https://stripe.com/docs/payments/payment-intents/quickstart#creating-with-manual-confirmation

The documentation says:

> Note that if your API version is before 2019-02-11, 'requires_action'
   appears as 'requires_source_action'.
> 

If I'm not mistaken, the package does not allow setting the api-version, see https://github.com/thephpleague/omnipay-stripe/issues/135

Meaning that, by default your code will use the version 2018-09-24 and because of this, `isRedirect` is always false because the value of `$this->getStatus()` is in my view "requires_source_action"

Reply:
> Meaning that, by default your code will use the version 2018-09-24 

The code will use whatever API version you've set in the dashboard and adding functionality to set API version per request is definitely out of scope for this pull request.

Reply:
To not break other payment methods, i would like to set the version explicitly only for credit card payments where I need to support 3DS. 

But I understand that this is out of scope here.

Reply:
Could this check for both? Just thinking in case people are using an older version of the API?

Reply:
@andris-sevcenko is it not possible to change 

```if ($this->getStatus() === 'requires_action') {``` to

```if ($this->getStatus() === 'requires_action' || $this->getStatus() === 'requires_source_action') {```